src/core/packages/saved-objects/base-server-internal/src/model_version/model_version_from_mappings.ts

```diff
@@ -51,16 +56,23 @@ export const getVirtualVersionsFromMappingMeta = ({
   meta,
   source,
   knownTypes,
+  minimumVirtualVersion,
 }: GetModelVersionsFromMappingMetaOpts): VirtualVersionMap | undefined => {
   const indexVersions = source === 'mappingVersions' ? meta.mappingVersions : meta.docVersions;
   if (!indexVersions) {
     return undefined;
   }
+
+  const minVersion = minimumVirtualVersion ? new SemVer(minimumVirtualVersion) : undefined;
   const typeSet = knownTypes ? new Set(knownTypes) : undefined;
 
   return Object.entries(indexVersions).reduce<VirtualVersionMap>((map, [type, rawVersion]) => {
     if (!typeSet || typeSet.has(type)) {
-      map[type] = assertValidVirtualVersion(rawVersion);
+      const validatedVersion = assertValidVirtualVersion(rawVersion);
+      map[type] =
+        minimumVirtualVersion && minVersion!.compare(validatedVersion) === 1
+          ? minimumVirtualVersion
+          : validatedVersion;
```

Comment:
With https://github.com/elastic/kibana/pull/219329 `index._meta.mappingVersions` can now have "downgraded" versions for those types that do not define `modelVersions:`. This can happen for customers that upgrade through 9.0.1 or 9.0.2.
When updating from any of the versions above, we don't want the logic to determine that those types have been updated, that's why we introduce this new `minimumVirtualVersion`.